crates/solvers/src/tests/mock/http.rs

```diff
@@ -1,22 +1,66 @@
 use std::{
+    fmt::{self, Debug, Formatter},
     net::SocketAddr,
     sync::{Arc, Mutex},
 };
 
-#[derive(Debug, Clone)]
+#[derive(Clone)]
+pub enum Path {
+    Any,
+    Exact(String),
+    Glob(glob::Pattern),
+}
+
+impl Path {
+    pub fn exact(s: impl ToString) -> Self {
+        Self::Exact(s.to_string())
+    }
+
+    pub fn glob(s: impl AsRef<str>) -> Self {
+        Self::Glob(glob::Pattern::new(s.as_ref()).unwrap())
+    }
+}
+
+impl PartialEq<Path> for String {
+    fn eq(&self, path: &Path) -> bool {
```

Comment:
Not blocking merge of PR but I think we should not use PartialEq for this. Using equality in the code below made sense when the path was a plain string, now that we're doing complex matching logic this is going to be confusing later. Would change it to `matches(&str) -> bool` and call that below.

Reply:
> Would change it to matches(&str) -> bool and call that below.

I had this, but then we can't use `assert_eq!` and need to manually build the panic message. Since it was for testing code, I thought it was fine to "abuse" the equality operator and `assert_eq!`.

In general, I agree with you though.

Reply:
You could have `assert_matches` which can build the panic message in one place. Anyway, no change required for merging.